packages/firestore/ios/Plugin/FirebaseFirestorePlugin.swift

```diff
@@ -348,17 +348,20 @@ public class FirebaseFirestorePlugin: CAPPlugin {
 
         let options = RemoveSnapshotListenerOptions(callbackId: callbackId)
 
-        implementation?.removeSnapshotListener(options)
-        call.resolve()
+        implementation?.removeSnapshotListener(options) {
+            call.resolve()
+        }
     }
 
     @objc override public func removeAllListeners(_ call: CAPPluginCall) {
         for (_, savedCall) in self.pluginCallMap {
             bridge?.releaseCall(savedCall)
         }
         self.pluginCallMap.removeAll()
-
-        implementation?.removeAllListeners()
-        super.removeAllListeners(call)
```

Comment:
@robingenz Unfortunately the superclass also takes responsibility for `call.resolve()` which wouldn't be correct when we are asynchronously needing to remove our implementation listeners. I've added `self.eventListeners?.removeAllObjects()` which is the only feature of the super call we're interested in, but I thought you might want to weigh in.